content/docs/guides/tls-encryption.md

```diff
@@ -0,0 +1,109 @@
+
+NOTE: Although TLS connections *to* Prometheus instances are not supported, TLS is supported for connections *from* Prometheus instances to [scrape targets](../prometheus/latest/configuration/configuration/#<tls_config>).
+
+## nginx example
+
+Let's say that you want to run a Prometheus instance behind an [nginx](https://www.nginx.com/) server available at the `my-proxy.com` domain (which you own), and for all Prometheus endpoints to be available via the `/prometheus` endpoint. The full URL for Prometheus' `/metrics` endpoint would thus be:
+
+```
+https://my-proxy.com/prometheus/metrics
+```
+
+Let's also say that you've generated the following using [OpenSSL](https://www.digitalocean.com/community/tutorials/openssl-essentials-working-with-ssl-certificates-private-keys-and-csrs) or an analogous tool:
+
+* an SSL certificate at `/root/certs/my-proxy.com/my-proxy.com.crt`
+* an SSL key at `/root/certs/my-proxy.com/my-proxy.com.key`
+
+You can generate a self-signed certificate and private key using this command:
+
+```bash
+mkdir -p /root/certs/my-proxy.com && cd /root/certs/my-proxy.com
+openssl req \
+  -x509 \
+  -newkey rsa:4096 \
+  -nodes \
+  -keyout my-proxy.com.key \
+  -out my-proxy.com.crt
+```
+
+Fill out the appropriate information at the prompts, and make sure to enter `my-proxy.com` at the `Common Name` prompt.
+
+## nginx configuration
+
+Below is an example [`nginx.conf`](https://www.nginx.com/resources/wiki/start/topics/examples/full/) configuration file. With this configuration, nginx will:
+
+* enforce TLS encryption using your provided certificate and key
+* proxy all connections to the `/prometheus` endpoint to a Prometheus server running on the same host (while removing the `/prometheus` from the URL)
+
+```conf
+events {}
+
+http {
+    server {
+        listen              443 ssl;
+        server_name         my-proxy.com;
+        ssl_certificate     /root/certs/my-proxy.com/my-proxy.com.crt;
+        ssl_certificate_key /root/certs/my-proxy.com/my-proxy.com.key;
+
+        location /prometheus {
+          proxy_pass http://localhost:9090/;
+        }
+    }
+}
+```
+
+Start nginx as root (since nginx will need to bind to port 443):
+
+```bash
+sudo nginx -c /usr/local/etc/nginx/nginx.conf
+```
+
+NOTE: This example uses `/usr/local/etc/nginx` as the location of the nginx configuration file, but this will vary based on the installation. Other [common nginx config directories](http://nginx.org/en/docs/beginners_guide.html) include `/usr/local/nginx/conf` and `/etc/nginx`.
+
+If nginx is already running, restart it with the new configuration:
```

Comment:
This is also something that varies across distros, I'd exclude it for simplicity.